### PR TITLE
Correct application of rotations to coordinates in `WeightedRigidAlign` within `alphafold3.py`

### PR DESCRIPTION
**Pull request by amorehead**
* Also adds a check to see if the rotations returned by `WeightedRigidAlign` are unique (thanks @pytorch3d!)
* Lastly, updates the unit test for `WeightedRigidAlign` to more carefully test the correctness of both `WeightedRigidAlign` and `CentreRandomAugmentation`

### CODE REVIEW

File: alphafold3_pytorch/alphafold3.py [hunk 1]
```diff
@@ -2339,7 +2354,7 @@ def forward(self, coords: Float['b n 3']) -> Float['b n 3']:
         translation_vector = rearrange(translation_vector, 'b c -> b 1 c')
 
         # Apply rotation and translation
-        augmented_coords = einsum(centered_coords, rotation_matrix, 'b n i, b i j -> b n j') + translation_vector
+        augmented_coords = einsum(centered_coords, rotation_matrix, 'b n i, b j i -> b n j') + translation_vector
```
Comment (amorehead) [inline alphafold3_pytorch/alphafold3.py:2357]: @lucidrains, note that I had to switch the axes `i` and `j` here to correctly rotate `centered_coords`. I believe we are using the `b i j` einsum rotation syntax in other places of the code, and (if they are similar to this use case) they may be incorrect as well.

Comment (lucidrains) [reply alphafold3_pytorch/alphafold3.py:2357]: nice, want to check those other places while you are at it?

your tests are failing though

Comment (amorehead) [reply alphafold3_pytorch/alphafold3.py:2357]: Whew... turns out I originally missed of these axes swaps lol when I was copying over the changes from my fork to `main`. Fixed now, and I'll start looking for all other instances of `b n i, b i j` rotations in the code.

Comment (amorehead) [reply alphafold3_pytorch/alphafold3.py:2385]: It looks like I was mistaken. `WeightedRigidAlign` and `CentreRandomAugmentation` are the only places in the code where rotations are applied to coordinates. I believe we are good to go now

**Pull request merged**